app/confirm/components/ConfirmSubscriptionForm.tsx

```diff
@@ -73,19 +71,47 @@ export function ConfirmSubscriptionForm({
     }
   };
 
+  const getSatoshisForInterval = (interval: string) => {
+    // Calculate the number of satoshis based on the desired interval
+    switch (interval) {
+      case "daily":
+        return 1; // Adjust this as needed
+      case "weekly":
+        return 7; // Adjust this as needed
+      case "monthly":
+        return 30; // Adjust this as needed
+      default:
+        return 1; // Default to daily
+    }
+  };
+
   const onSubmit = handleSubmit(async (data) => {
     if (!data.nostrWalletConnectUrl) {
       toast.error("Please link your wallet");
       return;
     }
+
+    const selectedInterval = data.sleepDuration;
+
+    // Enforce that the interval should not be more than 1 year (365 days)
+    if (selectedInterval === "yearly") {
```

Comment:
please use the duration in milliseconds. Otherwise this will let through e.g. 100 weeks or 1000000000 hours